File: climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int dp[50]={};
    int climbStairs(int n) {
        if(n==1) return 1;
        else if(n==2) return 2;
        else
        {
            int one,two;
            if(dp[n-2]!=0) one =dp[n-2];
            else{ 
                one =climbStairs(n-1);
                dp[n-2]=one;
            }
            if(dp[n-3]!=0) two =dp[n-3];
            else{
                two=climbStairs(n-2);
                dp[n-3]=two; 
                }
            dp[n-1]=one + two;
            return dp[n-1];
        }
    }
};

File: climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n==1) return 1;
        if(n==2) return 2;
        int thisOne=1;
        int nextOne=2;
        int temp=0;
        for(int i=2;i<n;i++){
            temp=thisOne+nextOne;
            thisOne=nextOne;
            nextOne=temp;
        }
        return nextOne;
    }
}